app/src/main/res/layout/fragment_marketplace.xml

```diff
@@ -69,38 +62,85 @@
-                tools:listitem="@layout/tabs_item"
+                android:overScrollMode="never"
+                android:paddingStart="22dp"
                 app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
                 app:layout_constraintEnd_toEndOf="parent"
                 app:layout_constraintStart_toStartOf="parent"
-                app:layout_constraintTop_toTopOf="parent" />
+                app:layout_constraintTop_toTopOf="parent"
+                tools:itemCount="10"
+                tools:listitem="@layout/item_tabs" />
 
             <androidx.recyclerview.widget.RecyclerView
                 android:id="@+id/tab_details_recycler"
                 android:layout_width="match_parent"
                 android:layout_height="0dp"
-                android:layout_marginStart="22dp"
-                android:layout_marginTop="26dp"
                 android:orientation="vertical"
-                app:layoutManager="androidx.recyclerview.widget.LinearLayoutManager"
+                android:overScrollMode="never"
+                android:paddingHorizontal="22dp"
+                android:paddingTop="20dp"
+                app:layoutManager="androidx.recyclerview.widget.GridLayoutManager"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintHorizontal_bias="0.0"
+                app:layout_constraintStart_toStartOf="parent"
                 app:layout_constraintTop_toBottomOf="@id/tabs_recycler"
-                app:layout_constraintBottom_toBottomOf="parent"/>
+                app:layout_constraintVertical_bias="0.0"
+                app:spanCount="2"
+                tools:itemCount="7"
+                tools:listitem="@layout/item_tab_details" />
+
+            <include
+                android:id="@+id/placeholder_tab_details"
+                layout="@layout/placeholder_no_items"
+                android:layout_width="match_parent"
+                android:layout_height="0dp"
+                android:layout_marginTop="20dp"
+                android:background="@color/very_light_pink"
+                android:visibility="gone"
+                app:layout_constraintBottom_toBottomOf="parent"
+                app:layout_constraintEnd_toEndOf="parent"
+                app:layout_constraintStart_toStartOf="parent"
+                app:layout_constraintTop_toBottomOf="@id/tabs_recycler" />
 
         </androidx.constraintlayout.widget.ConstraintLayout>
 
+        <ProgressBar
+            android:id="@+id/progress_indicator"
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            app:layout_constraintBottom_toBottomOf="parent"
+            app:layout_constraintEnd_toEndOf="parent"
+            app:layout_constraintStart_toStartOf="parent"
+            app:layout_constraintTop_toTopOf="parent" />
+
+        <include
```

Comment:
why do you repeat include layout/placeholder_no_items

Reply:
placeholder -> one if no category is retrieved and one for if category list is empty (like category number 2 "Horse Trading")

Reply:
👍